TypeCobol.LanguageServer/Completion Factory/CompletionFactoryHelpers.cs

```diff
@@ -224,5 +230,89 @@ public static CompletionItem CreateCompletionItemForIndex(Compiler.CodeElements.
             return new CompletionItem(display) {insertText = index.Name, kind = CompletionItemKind.Variable};
         }
+
+        public static Case GetTextCase(string tokenText)
+        {
+            if (string.IsNullOrEmpty(tokenText)) return Case.Lower;
+            // check if upper case
+            bool isUpper = true;
+            foreach (char c in tokenText)
+            {
+                if (char.IsLower(c))
+                {
+                    isUpper = false;
+                    break;
+                }
+            }
+
+            if (isUpper) return Case.Upper;
+            // check if Camel case; ex. CamlSample
+            bool isCamel = false;
+            int iUpperMem = -1, count = tokenText.Length;
+            for (int i = 0; i < count; i++)
+            {
+                if (i == 0)
+                {
+                    // first letter
+                    if (char.IsLower(tokenText[i])) break;
+                    iUpperMem = 0;
+                    isCamel = true;
+                }
+                else
+                {
+                    if (char.IsUpper(tokenText[i]))
+                    {
+                        if (iUpperMem == (i - 1))
+                        {
+                            // 2 upper case characters one behind the other
+                            isCamel = false;
+                            break;
+                        }
+                        iUpperMem = i;
+                    }
+                }
+            }
+
+            if (isCamel && iUpperMem == count - 1) isCamel = false; // last char is upper
+            return (isCamel) ? Case.Camel : Case.Lower;
+        }
+
+        private static readonly Dictionary<ParameterDescription.PassingTypes, string> _UpperParams = new Dictionary<ParameterDescription.PassingTypes, string>()
+        {
+            { ParameterDescription.PassingTypes.Input, "INPUT"},
+            { ParameterDescription.PassingTypes.Output, "OUTPUT" },
+            { ParameterDescription.PassingTypes.InOut, "IN-OUT" }
+        };
+        private static readonly Dictionary<ParameterDescription.PassingTypes, string> _CamelParams = new Dictionary<ParameterDescription.PassingTypes, string>()
+        {
+            { ParameterDescription.PassingTypes.Input, "Input"},
+            { ParameterDescription.PassingTypes.Output, "Output" },
+            { ParameterDescription.PassingTypes.InOut, "In-Out" }
+        };
+        private static readonly Dictionary<ParameterDescription.PassingTypes, string> _LowerParams = new Dictionary<ParameterDescription.PassingTypes, string>()
+        {
+            { ParameterDescription.PassingTypes.Input, "input"},
+            { ParameterDescription.PassingTypes.Output, "output" },
+            { ParameterDescription.PassingTypes.InOut, "in-out" }
+        };
+
+        public static Dictionary<ParameterDescription.PassingTypes, string> GetParamsWithCase(Case textCase)
```

Comment:
Could be refactored as `public static string GetPassingTypeString(PassingTypes passingType, Case textCase)`. This would remove the need to pass the Dictionary in the other methods.
To be discussed because as you pointed out, this would require to check for the dictionary selection at each call.

Reply:
It would require to check for the dictionary at each call; solution with a Dictionary determined only one time is faster at execution